### ISSUE TITLE
Metadata and Content-Disposition

### ISSUE THREAD

**Issue by KevinLu** [ikhsanalatsary/multer-sharp-s3#38]
Will this library support Metadata and Content-Disposition? 

I've tried adding Content-Disposition like this:

```JavaScript
var upload = multer({
    storage: multerS3({
        s3: s3,
        Bucket: s3Bucket,
        ContentType: multerS3.AUTO_CONTENT_TYPE,
        ContentDisposition: 'attachment',
        Key: (req, file, cb) => {
            cb(null, file.originalname);
        }
    }),
    fileFilter: mediaFilter
});
```

but in my S3 bucket, the metadata Content-Disposition does not get added.

[S3 Upload Property](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#upload-property) has Content-Disposition as one of its properties.

Looking forward to this.

**Comment by rramiachraf** [comment]
How did you guys used the AUTO_CONTENT_TYPE from 'multer-s3', i can't get it done correctly, ContentType is throwing a typescript error and when i silence it, it throws this error "Expected params.ContentType to be a string"

**Comment by KevinLu** [comment]
> 
> 
> How did you guys used the AUTO_CONTENT_TYPE from 'multer-s3', i can't get it done correctly, ContentType is throwing a typescript error and when i silence it, it throws this error "Expected params.ContentType to be a string"

Sorry, when I tested this further. I got the same error as you. AUTO_CONTENT_TYPE is not implemented in multer-sharp-s3 so there's no hope of getting that working. Did you ever find a fix for this?

**Comment by rramiachraf** [comment]
Unfortunately not, I just abandoned both multer-s3 and multer-sharp-s3 and I just used raw aws-sdk with sharp and it's even better.

**Comment by KevinLu** [comment]
> 
> 
> Unfortunately not, I just abandoned both of multer-s3 and multer-sharp-s3 and I just used raw aws-sdk with sharp and it's even better.

Yup, that's what I did do. AWS sdk on its own with Sharp seems to work a lot better.

**Comment by ikhsanalatsary** [comment]
Hi @KevinLu, thanks for addressing this issue. for `contentType`, this lib has a default value if you don't specify it will check based on your file type. for the `ContentDisposition`, this lib not supported it yet. I will check if I have spare time.

**Comment by KevinLu** [comment]
> 
> 
> Hi @KevinLu, thanks for addressing this issue. for `contentType`, this lib has a default value if you don't specify it will check based on your file type. for the `ContentDisposition`, this lib not supported it yet. I will check if I have spare time.

Thanks for the response, let me know if you've gotten it sorted out. I'd love to give this package another try.

**Comment by ozgrozer** [comment]
@rramiachraf @KevinLu Can you guys share an example code? I'd like to go with aws-sdk and sharp too. Since this package is not maintained anymore.

**Comment by rramiachraf** [comment]
@ozgrozer here you go man, https://github.com/rramiachraf/buffalo/blob/main/backend/src/rest/updatePoster.ts

**Comment by ozgrozer** [comment]
@rramiachraf Thank you for the code. It helped me a lot.

**Comment by ikhsanalatsary** [comment]
Thank you for addressing this issue. I think there's a missing in this lib 😃 . If you don't mind and can fix this. PR is always welcome. FYI, I just published v.0.2.2.
